crates/core/src/host/instance_env.rs

```diff
@@ -33,6 +34,53 @@ pub struct TxSlot {
     inner: Arc<Mutex<Option<MutTxId>>>,
 }
 
+#[derive(Default)]
+struct ChunkedWriter {
+    chunks: Vec<Box<[u8]>>,
+    scratch_space: Vec<u8>,
+}
+
+impl BufWriter for ChunkedWriter {
+    fn put_slice(&mut self, slice: &[u8]) {
+        self.scratch_space.extend_from_slice(slice);
+    }
+}
+
+impl ChunkedWriter {
+    pub fn force_flush(&mut self) {
```

Comment:
Doc comments on `force_flush` and `flush` and I'm happy.

Reply:
Ah right, I was kind of hoping that #420 will be merged and I'll be able to get rid of separate `force_flush` altogether. I'll add to both anyway.